wikimedia/wikipedia-ios
# AaaLD - Event logging

**Fixes Phabricator ticket:** 
https://phabricator.wikimedia.org/T262616

### Notes

Added Event Logging to the Article as a Living Document screens using the legacy logging system. Schema and deck with screenshots are linked in the Phab task. It's worth noting that Shay will QA and confirm all of these in the back end once they are in a TestFlight build, so for this code review we don't have to get too in the weeds on what exactly is sent.

### Test Steps

1. Fresh install app. In onboarding toggle on the sending of analytics data on the last page.
2. Perform various actions in the schema deck. You should see something like `EventLoggingService: 0x60000337dd20 <x-coredata:///WMFEventRecord/t2B6FC470-29D4-46F2-B143-AE3D8F2D0B4A7> recorded!` in the console with each event you trigger.
3. Watch the network via a proxy. You will see dumps every so often of events that start with `https://meta.wikimedia.org/beacon/event...`.
4. Take that event url and decode it through a [url decoder](https://meyerweb.com/eric/tools/dencoder/). You should see events with the schema value of `MobileWikiAppiOSLivingDoc`. Confirm the `action` field, `position` and `type` fields (if applicable) match what's expected.
